src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String CSV_FILE_SEPERATOR = ",";
+    private static final String OPER_TYPE_SUPPLY = "supply";
+    private static final String OPER_TYPE_BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readDataFromFile(fromFileName);
+
+        String report = createReport(dataFromFile);
+
+        writeDataToFile(report, toFileName);
+    }
+
+    private String readDataFromFile(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+
+            String lineFromFile = bufferedReader.readLine();
+            while (lineFromFile != null) {
+                stringBuilder.append(lineFromFile).append(System.lineSeparator());
```

Comment:
Concatenating strings inside a loop can be inefficient because each concatenation creates a new object. Consider using `StringBuilder.append()` instead.